src/main/java/com/amihaiemil/versioneye/MkMe.java

```diff
@@ -43,23 +46,33 @@
     private MkServer server;
     
     /**
-     * Mock Authenticated user.
+     * Authenticated user.
      */
-    private Authenticated authenticated;
+    private String username;
     
     /**
      * Ctor.
      * @param server VersionEye server storage.
-     * @param authenticated Authenticated user.
+     * @param username The authenticated user's name.
      */
-    MkMe(final MkServer server, final Authenticated authenticated) {
+    MkMe(final MkServer server, final String username) {
         this.server = server;
-        this.authenticated = authenticated;
+        this.username = username;
     }
     
     @Override
     public Authenticated about() throws IOException {
-        return this.authenticated;
+        final JsonArray online = this.server.storage().build()
+            .getJsonArray("authenticated");
+        for(int idx = 0; idx < online.size(); idx++) {
+            final JsonObject user = online.getJsonObject(idx);
+            if (user.getJsonObject(this.username) != null) {
+                return new MkJsonAuthenticated(user.getJsonObject(this.username));
+            }
+        }
+        throw new IllegalStateException (
```

Comment:
@SherifWaly I moved here the exception for unauthenticated users. It looks better and makes the most sense. When you query the versionEye server it first has to look what it holds, and if it doesn't find any user with the given username, then it throws an exception :D